src/controllers/v1/match2/index.ts

```diff
@@ -312,16 +288,47 @@ export class Match2Controller extends Controller {
       return transaction
     }
 
+    const acceptRematch = async () => {
+      const ownsNewDemandB = demandB.owner === selfAddress
+      if (!ownsDemandA && !ownsNewDemandB) throw new BadRequest(`You do not own an acceptable demand`)
+      if (match2.state === 'proposed') return acceptAB()
+      const [oldDemandB]: DemandRow[] = await this.db.getDemand(oldMatch2.demandB)
+
+      const extrinsic = await this.node.prepareRunProcess(
+        rematch2AcceptFinal({
+          match2: oldMatch2,
+          demandA,
+          demandB: oldDemandB,
+          newDemandB: demandB,
+          newMatch2: match2,
+        })
+      )
+
+      const [transaction] = await this.db.insertTransaction({
+        transaction_type: 'accept',
+        api_type: 'match2',
+        local_id: match2.id,
+        state: 'submitted',
+        hash: extrinsic.hash.toHex(),
+      })
+
+      this.node.submitRunProcess(extrinsic, this.db.updateTransactionState(transaction.id))
+      return transaction
+    }
+
+    const [oldMatch2]: Match2Row[] = match2.replaces ? await this.db.getMatch2(match2.replaces) : []
+    if (match2.replaces) return acceptRematch()
```

Comment:
the control flow here is pretty ugly. Would it not be better to update the switch statement on ln 331 to return either `acceptFinal` or `acceptRematch` depending on whether this is a rematch or not. I just found it quite hard to reason about what the different checks are doing

Reply:
yes, that has been updated.